src/Client/DurableStorage/FileStorage.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Tuf\Client\DurableStorage;
+
+/**
+ * Class FileStorage
+ *
+ * A simple implementation of \ArrayAccess using the filesystem.
+ * Applications might want to provide an alternative implementation with
+ * better performance and error handling.
+ *
+ * @TODO Add tests for this class.
+ */
+class FileStorage implements \ArrayAccess
+{
+    /**
+     * @var string $basePath
+     *   The path on the filesystem to this durable storage's files.
+     */
+    protected $basePath;
+
+    public function __construct(string $basePath)
+    {
+        if (! is_dir($basePath)) {
+            throw new \RuntimeException("Cannot initialize filesystem local state: \"$basePath\" is not a directory.");
+        }
+
+        $this->basePath = $basePath;
+    }
+
+    protected function offsetToPath($offset)
+    {
+        return $this->basePath . DIRECTORY_SEPARATOR . $offset;
+    }
```

Comment:
So this helper method is protected (which I missed before) and is the only method not required to implement `\ArrayAccess`.

It needs a docblock. I also still think we need a different method name because I don't know what it means from reading the name. Maybe just `pathWithBasePath()` or something?